connector/github/github.go

```diff
@@ -575,7 +580,21 @@ func (c *githubConnector) userEmail(ctx context.Context, client *http.Client) (s
 			}
 
 			if email.Verified && email.Primary {
-				return email.Email, nil
+				primaryEmail = email
+			}
+
+			if c.preferredEmailDomain != "" {
+				components := strings.Split(email.Email, "@")
+				if email.Verified && components[1] == c.preferredEmailDomain {
+					return email.Email, nil
+				}
+
+				// fallback to primary email if there is no preferred-domain email
+				if idx == len(emails)-1 {
+					return primaryEmail.Email, nil
```

Comment:
If both primary email and email with a preferred domain were missing, an error should be returned.